datapackage/resource.py

```diff
@@ -312,38 +310,21 @@ def _parse_inline_data(self):
 
         return data
 
+    def _iter_from_tabulator(self, table, schema):
+        if schema is not None:
+            schema = jsontableschema.model.SchemaModel(schema)
+        for keyed_row in table.iter(keyed=True):
+            if schema is not None:
+                for field in schema.fields:
+                    try:
+                        keyed_row[field['name']] = schema.cast(
+                            field['name'], keyed_row[field['name']])
```

Comment:
Minor suggestion: define `column = field['name']` and use it instead of repeating `field['name']`. You could even use `col` instead to make it possible to write this statement in a single line.